client/desktop/lib/cookie-auth/index.js

```diff
@@ -70,49 +67,62 @@ function parseCookie( cookieStr ) {
 	}, cookie );
 }
 
-function setSessionCookies( window, onComplete ) {
-	return function ( response ) {
-		let cookieHeaders = response.headers[ 'set-cookie' ];
+function setSessionCookies( window, authorizeResponse ) {
+	return new Promise( ( resolve ) => {
+		let cookieHeaders = authorizeResponse.headers[ 'set-cookie' ];
 		let count = 0;
 		if ( ! Array.isArray( cookieHeaders ) ) {
 			cookieHeaders = [ cookieHeaders ];
 		}
 
 		count = cookieHeaders.length;
 
-		cookieHeaders.map( parseCookie ).forEach( function ( cookie ) {
+		if ( count === 0 ) {
+			return resolve( true );
+		}
+
+		cookieHeaders.forEach( async function ( cookieStr ) {
+			const cookie = parseCookie( cookieStr );
 			cookie.url = 'https://wordpress.com/';
-			if ( cookie.httponly ) {
+			if ( cookie.HttpOnly || cookie.httpOnly || cookie.httponly ) {
 				cookie.session = true;
 			}
-			window.webContents.session.cookies.set( cookie, function () {
-				count--;
-				if ( count === 0 ) {
-					if ( onComplete ) onComplete();
-				}
-			} );
+			try {
+				await window.webContents.session.cookies.set( cookie );
+			} catch ( e ) {
+				const { value, ...logCookie } = cookie; // don't log sensitive "value" field
+				log.error( `Failed to set session cookie (${ e.message }): `, logCookie );
+			}
+			count--;
+			if ( count === 0 ) {
+				return resolve( true );
+			}
 		} );
-	};
+	} );
 }
 
 function auth( window, onAuthorized ) {
-	let currentRequest;
+	ipc.on( 'user-auth', async function ( _, user, token ) {
+		log.info( `Handling 'user-auth' IPC event, setting session cookies...` );
 
-	ipc.on( 'user-auth', function ( event, user, token ) {
 		if ( user && user.data ) {
 			const userData = user.data;
-			if ( currentRequest && currentRequest.username === userData.username ) {
-				// already authing
-				return;
+
+			try {
+				const response = await authorize( userData.username, token );
+				await setSessionCookies( window, response );
+				onAuthorized();
+
+				log.info( 'Session cookies set.' );
+			} catch ( e ) {
+				log.error( 'Failed to set session cookies: ', e );
 			}
-			currentRequest = authorize( userData.username, token ).on(
-				'response',
-				setSessionCookies( window, onAuthorized )
-			);
 		} else {
-			// retrieve all cookies
+			log.info( 'No user data, clearing session cookies...' );
+
 			window.webContents.session.cookies.get( {}, function ( e, cookies ) {
```

Comment:
`session.cookies.get` also returns a promise in recent Electrons, so the callback can be removed in favor of an `await`.